### DISCUSSION TITLE
How to get reproducibility ?

### DISCUSSION THREAD

**Discussion by filot997** [jamesdolezal/slideflow#342] [Q&A]
When I try to split the data, I get different data each time by using following codes.

`train_dataset, val_dataset = dataset.split(labels="category", val_fraction=0.2)`

So I tried to fix the seed as follow, But it failed.

`random.seed(seed)`
`os.environ['PYTHONHASHSEED'] = str(seed)`
`np.random.seed(seed)`
`torch.manual_seed(seed)`
`torch.cuda.manual_seed(seed)`
`torch.backends.cudnn.deterministic = True`
`torch.backends.cudnn.benchmark = False`

is there any way to solve this problem ?

**Comment by jamesdolezal** [comment]
Thanks for raising this question.

To log splits for reproducibility and future use, pass a path to a JSON file to the `splits` argument of `Dataset.split()`. The training/validation split will be saved there upon first use, and subsequent calls will re-use the same split. E.g.:

```python
def get splits():
  return dataset.split(
    labels="category", 
    val_fraction=0.2,
    splits='my_splits.json'
  )

# First call to this function will log the split
train_dataset, val_dataset = get_splits()

# The second call will return the same splits,
# loaded from the JSON file.
train_dataset, val_dataset = get_splits()
```

This is briefly mentioned in our documentation [here](https://slideflow.dev/datasets_and_val/#creating-a-split) and [here](https://slideflow.dev/dataset/#slideflow.Dataset.split), but we'll add a couple of sentences to make this clearer.
